src/software/backend/simulation/simulator_ball.cpp

```diff
@@ -0,0 +1,54 @@
+#include "software/backend/simulation/simulator_ball.h"
+
+std::weak_ptr<PhysicsBall> SimulatorBallSingleton::physics_ball_weak_ptr =
```

Comment:
```suggestion
SimulatorBallSingleton::physics_ball_weak_ptr =
```

Reply:
That's just how C++ likes static variables initialized. Without the type I get the following error

> software/backend/simulation/simulator_ball.cpp:3:25: error: C++ requires a type specifier for all declarations
SimulatorBallSingleton::physics_ball_weak_ptr =